### PR TITLE
Add ability to split cell

### PR DESCRIPTION
**Pull request by captainsafia**
Split cell soup!

### CODE REVIEW

**Comment by captainsafia** [comment]
OK! I feel like this can be merged because I don't want it to be a huge PR.

The next steps for this are.
- Fix bug in commutable and make it so that the newly created cell is added after the current one not at the end of the file
- Get the position of the cursor and use it when splitting, this will be made easier once I get context menu working

**Comment by captainsafia** [comment]
cc: @rgbkrk @kenwheeler

File: src/notebook/reducers/document.js [hunk 1]
```diff
@@ -138,6 +138,10 @@ export default handleActions({
     const { id } = action;
     return state.update('notebook', (notebook) => commutable.clearCellOutput(notebook, id));
   },
+  [constants.SPLIT_CELL]: function splitCell(state, action) {
+    const { id, position } = action;
+    return state.update('notebook', (notebook) => commutable.splitCell(notebook, id, position));
```
Comment (rgbkrk) [inline src/notebook/reducers/document.js:143]: It's ok to do the logic here for now and _then_ add it to commutable.

**Comment by rgbkrk** [comment]
I'll merge it as is and look forward to you addressing the rest.

**Pull request merged**